supporting-blog-content/evaluating-search-relevance-part-1/Retrieve and rerank.ipynb

```diff
@@ -0,0 +1,801 @@
+   "id": "a05f9722-ebc1-43fc-9fa4-c50ef72ea287",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client = Elasticsearch(\n",
+    "    cloud_id=ELASTIC_CLOUD_ID,\n",
+    "    api_key=ELASTIC_API_KEY,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4549ab8a-6add-4a9d-a6c9-d1391de914a3",
+   "metadata": {},
+   "source": [
+    "### Test the client\n",
+    "\n",
+    "Before you continue, confirm that the client has connected with this test.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0336efb4-5d77-46e4-8d93-ef03b2de1b93",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client_info = client.info()\n",
+    "\n",
+    "f\"Successfully connected to cluster {client_info['cluster_name']} (version {client_info['version']['number']})\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87eeef16-c040-4760-9be6-517fc6eefbac",
+   "metadata": {},
+   "source": [
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4de131fe-e8ec-40a2-92aa-5765235f01a9",
+   "metadata": {},
+   "source": [
+    "## Helper functions\n",
+    "\n",
+    "In this section we define some helper functions to increase the readability of our code.\n",
+    "\n",
+    "Let's start with the functions that will handle the interaction with our Elastic Cloud deployment such as: \n",
+    "- creating an index\n",
+    "- storing the documents\n",
+    "- retrieving documents with BM25"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "073294e4-8893-4c0a-9e80-7f34f1ea81c2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_index(es_client: Elasticsearch, name: str, analyzer: str = \"english\"):\n",
+    "    \"\"\"\n",
+    "    Creating an index into our deployment\n",
+    "\n",
+    "    Args:\n",
+    "        `es_client`: An instance of a Python Elasticsearch client\n",
+    "        `analyzer`: A string identifier of the language analyzer to be used. By default we use `english`\n",
+    "            (more details at https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-lang-analyzer.html)\n",
+    "    Returns:\n",
+    "        None\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # we store `title` & `text` into separate fields and\n",
+    "    _mappings = {\n",
+    "        \"properties\": {\n",
+    "            \"title\": {\"type\": \"text\", \"analyzer\": analyzer},\n",
+    "            \"txt\": {\"type\": \"text\", \"analyzer\": analyzer},\n",
+    "        }\n",
+    "    }\n",
+    "\n",
+    "    # create an index with the specified name\n",
+    "    es_client.options(ignore_status=[400]).indices.create(\n",
+    "        index=name,\n",
+    "        settings={\"number_of_shards\": 1},\n",
+    "        mappings=_mappings,\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "def index_corpus(\n",
+    "    corpus: datasets.arrow_dataset.Dataset, index_name: str, es_client: Elasticsearch\n",
```

Comment:
Are you sure corpus is an arrow dataset by default ?

Reply:
Good catch. I will fix it